File: src/main/java/org/example/expert/domain/todo/service/LogService.java
package org.example.expert.domain.todo.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class LogService {
    private final LogRepository logRepository;

    @Transactional(propagation = Propagation.REQUIRES_NEW)
    public void saveLog(String message, String requestIp, String requestUser) {
        Log log = new Log(message, requestIp, requestUser);
        logRepository.save(log);
    }
}

File: src/main/java/org/example/expert/domain/todo/service/LogService.kt
package org.example.expert.domain.todo.service

import org.example.expert.domain.todo.entity.Log
import org.example.expert.domain.todo.repository.LogRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional

@Service
class LogService (
    private val logRepository: LogRepository
) {
    @Transactional(propagation = Propagation.REQUIRES_NEW)
    fun saveLog(message: String, requestIp: String, requestUser: String){
        val log = Log(message, requestIp, requestUser)
        logRepository.save(log)
    }
}